app/controllers/course/survey/responses_controller.rb

```diff
@@ -28,11 +28,22 @@ def edit
   end
 
   def update
-    @response.submit if params[:response][:submit]
-    if @response.update_attributes(response_update_params)
-      render_response_json
-    else
-      render json: { errors: @response.errors }, status: :bad_request
+    if params[:response][:unsubmit]
```

Comment:
Can I suggest that since `unsubmit` is rather different from `update`, we just implement it in a separate method? Otherwise we are trying to do too many things in this controller method. Understand some rework is required for the react side. Thoughts? @kxmbrian @allenwq